### PR TITLE
Centralize push notifications logic

### PR DESCRIPTION
**Pull request by dejan2k**
What              | Where/Who
------------------|----------------------------------------
JIRA Issue        | https://leanplum.atlassian.net/browse/LP-11297

### CODE REVIEW

File: Leanplum-SDK/Classes/Internal/Leanplum.m [hunk 1]
```diff
@@ -908,7 +902,7 @@ + (void)startWithUserId:(NSString *)userId
     params[LP_PARAM_INBOX_MESSAGES] = [self.inbox messagesIds];
     
     // Push token.
-    NSString *pushTokenKey = [Leanplum pushTokenKey];
+    NSString *pushTokenKey = [[LPPushNotificationsManager sharedManager].handler pushTokenKey];
```
Comment (e7mac) [inline Leanplum-SDK/Classes/Internal/Leanplum.m:905]: could we move the persistence logic to the manager as well?

File: Leanplum-SDK/Classes/Notifications/LPNotificationsHelper.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+//  Leanplum-iOS-SDK
+//
+//  Created by Dejan . Krstevski on 15.05.20.
+//
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface LPNotificationsHelper : NSObject
```
Comment (e7mac) [inline Leanplum-SDK/Classes/Notifications/LPNotificationsHelper.h:12]: how are the helper vs manager responsibilities split up?

File: Leanplum-SDK/Classes/Notifications/Push/LPPushNotificationsHandler.h [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#pragma clang diagnostic ignored "-Wstrict-prototypes"
+- (void)didRegisterUserNotificationSettings:(UIUserNotificationSettings *)notificationSettings;
+- (void)sendUserNotificationSettingsIfChanged:(UIUserNotificationSettings *)notificationSettings;
+#pragma clang diagnostic pop
+
+/**
+* Returns a push token using app ID, device ID, and user ID.
+*/
+- (NSString *)pushTokenKey;
```
Comment (e7mac) [inline Leanplum-SDK/Classes/Notifications/Push/LPPushNotificationsHandler.h:43]: is this the push token or the push token key to obtain from UserDefaults?

File: Leanplum-SDK/Classes/Notifications/LPNotificationsHelper.h (hunk 2 above)
Comment (dejan2k) [reply Leanplum-SDK/Classes/Notifications/LPNotificationsHelper.h:12]: was planning to do something like common manager for local and push.... I named it helper (not the best name), planning to refactor this

File: Leanplum-SDK/Classes/Notifications/Push/LPPushNotificationsHandler.h (hunk 3 above)
Comment (dejan2k) [reply Leanplum-SDK/Classes/Notifications/Push/LPPushNotificationsHandler.h:43]: push token key

File: Leanplum-SDK/Classes/Notifications/LPNotificationsConstants.h [hunk 4]
```diff
@@ -0,0 +1,24 @@
+//
+//  LPNotificationsConstants.h
+//  Leanplum-iOS-SDK
+//
+//  Created by Dejan . Krstevski on 13.05.20.
```
Comment (milos1290) [inline Leanplum-SDK/Classes/Notifications/LPNotificationsConstants.h:5]: Fix copyright header

Comment (dejan2k) [reply Leanplum-SDK/Classes/Notifications/LPNotificationsConstants.h]: fixed

File: Leanplum-SDK/Classes/Internal/Leanplum.m (hunk 1 above)
Comment (dejan2k) [reply Leanplum-SDK/Classes/Internal/Leanplum.m]: all logic for push token is moved into the manager

**Pull request merged**